src/io/ExportXML.cpp

```diff
@@ -196,6 +244,11 @@ void ExportXML::exportData(
     }
     outFile << '\n'
             << "            </DataArray>\n";
+    if (hasGradient && dataDimensions > 1) {
+      exportGradient(data, spaceDim, outFile);
+    } else if (hasGradient && dataDimensions == 1) {
+      exportGradient(data, 1, outFile);
```

Comment:
Is the `spaceDim` argument really the spatial dimension? We usually don't support `dim=1` and you also named the variable in the function `spaceDim`. I guess it is rather the data dimension?! If that's the case, couldn't we omit the branch here and just write
```c++
    if (hasGradient) {
      exportGradient(data, dataDimensions, outFile);
    }
```
?

Reply:
Yes it is dataDimension